concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/AbstractLimiter.java

```diff
@@ -77,19 +105,47 @@ protected AbstractLimiter(Builder<?> builder) {
         this.limitAlgorithm = builder.limit;
         this.limit = limitAlgorithm.getLimit();
         this.limitAlgorithm.notifyOnChange(this::onNewLimit);
-
+        if (builder instanceof BypassLimiterBuilder) {
+            this.bypassResolvers = ((BypassLimiterBuilder) builder).bypassResolvers;
+        }
         builder.registry.gauge(MetricIds.LIMIT_NAME, this::getLimit);
         this.successCounter = builder.registry.counter(MetricIds.CALL_NAME, ID_TAG, builder.name, STATUS_TAG, "success");
         this.droppedCounter = builder.registry.counter(MetricIds.CALL_NAME, ID_TAG, builder.name, STATUS_TAG, "dropped");
         this.ignoredCounter = builder.registry.counter(MetricIds.CALL_NAME, ID_TAG, builder.name, STATUS_TAG, "ignored");
         this.rejectedCounter = builder.registry.counter(MetricIds.CALL_NAME, ID_TAG, builder.name, STATUS_TAG, "rejected");
+        this.bypassCounter = builder.registry.counter(MetricIds.CALL_NAME, ID_TAG, builder.name, STATUS_TAG, "bypassed");
+    }
+
+    protected boolean shouldBypass(ContextT context){
+        return bypassResolvers.stream().anyMatch(bypassResolver -> bypassResolver.test(context));
```

Comment:
We should avoid using streams on the hot path, it tends to cause performance issues. Let's swap this to a `for:each` loop or just chain the predicates together (aggregated by `Predicate::or`).

Reply:
Replaced with chained predicates

Reply:
That's much easier 😁 . I added one suggestion to optimize away the default.